internal/access/identity.go

```diff
@@ -61,6 +78,15 @@ func DeleteIdentity(c *gin.Context, id uid.ID) error {
 		return fmt.Errorf("cannot delete self: %w", internal.ErrForbidden)
 	}
 
+	internalIdentity, err := isInternalIdentity(c, id)
```

Comment:
What does this guard against? Is it just for preventing internal identities from making external requests for their own data?

Reply:
This is to prevent the user from removing internal identities as that will have unexpected consequences